import "maidsafe/shared/maidsafe_service_messages.proto";

package maidsafe;

service MaidsafeService {
  // chunk operations
  rpc StorePrep (StorePrepRequest) returns (StorePrepResponse);
  rpc StoreChunk (StoreChunkRequest) returns (StoreChunkResponse);
  rpc GetChunk (GetChunkRequest) returns (GetChunkResponse);
  rpc CheckChunk (CheckChunkRequest) returns (CheckChunkResponse);
  rpc DeleteChunk (DeleteChunkRequest) returns (DeleteChunkResponse);
  rpc ValidityCheck (ValidityCheckRequest) returns (ValidityCheckResponse);
  rpc SwapChunk (SwapChunkRequest) returns (SwapChunkResponse);
  rpc CacheChunk (CacheChunkRequest) returns (CacheChunkResponse);

  // chunk info operations
  rpc GetChunkReferences (GetChunkReferencesRequest) returns (GetChunkReferencesResponse);
  rpc AddToWatchList (AddToWatchListRequest) returns (AddToWatchListResponse);
  rpc RemoveFromWatchList (RemoveFromWatchListRequest) returns (RemoveFromWatchListResponse);
  rpc AddToReferenceList (AddToReferenceListRequest) returns (AddToReferenceListResponse);

  // account operations
  rpc AmendAccount (AmendAccountRequest) returns (AmendAccountResponse);
  rpc ExpectAmendment (ExpectAmendmentRequest) returns (ExpectAmendmentResponse);
  rpc AccountStatus (AccountStatusRequest) returns (AccountStatusResponse);

  // vault synchronisation operations
  rpc GetSyncData (GetSyncDataRequest) returns (GetSyncDataResponse);
  rpc GetAccount (GetAccountRequest) returns (GetAccountResponse);
  rpc GetChunkInfo (GetChunkInfoRequest) returns (GetChunkInfoResponse);
  rpc GetBufferPacket (GetBufferPacketRequest) returns (GetBufferPacketResponse);

  // buffer packet operations
  rpc CreateBP(CreateBPRequest) returns (CreateBPResponse);
  rpc ModifyBPInfo(ModifyBPInfoRequest) returns (ModifyBPInfoResponse);
  rpc GetBPMessages(GetBPMessagesRequest) returns (GetBPMessagesResponse);
  rpc AddBPMessage(AddBPMessageRequest) returns (AddBPMessageResponse);
  rpc GetBPPresence(GetBPPresenceRequest) returns (GetBPPresenceResponse);
  rpc AddBPPresence(AddBPPresenceRequest) returns (AddBPPresenceResponse);
}

service VaultRegistration {
  rpc SetLocalVaultOwned(SetLocalVaultOwnedRequest) returns (SetLocalVaultOwnedResponse);
  rpc LocalVaultOwned(LocalVaultOwnedRequest) returns (LocalVaultOwnedResponse);
}
